src/registry/customerProfile/index.tsx

```diff
@@ -0,0 +1,86 @@
+import type {CustomerProfileComponentSchema, DigitalAddress} from '@open-formulieren/types';
+import type {FormikErrors} from 'formik';
+import {useFormikContext} from 'formik';
+import {useId} from 'react';
+
+import {ValidationErrors} from '@/components/forms';
+import Fieldset from '@/components/forms/Fieldset';
+import HelpText from '@/components/forms/HelpText';
+import Tooltip from '@/components/forms/Tooltip';
+import {useFieldConfig} from '@/hooks';
+import type {RegistryEntry} from '@/registry/types';
+
+import './CustomerProfile.scss';
+import {useDigitalAddresses} from './digitalAddresses';
+import isEmpty from './empty';
+import {DigitalAddressFields} from './subFields';
+import getValidationSchema from './validationSchema';
+
+export interface FormioCustomerProfileProps {
+  componentDefinition: CustomerProfileComponentSchema;
+}
+
+export const FormioCustomerProfile: React.FC<FormioCustomerProfileProps> = ({
+  componentDefinition: {key: name, label, tooltip, description, validate, digitalAddressTypes},
+}) => {
+  const {getFieldMeta} = useFormikContext();
+  const id = useId();
+  name = useFieldConfig(name);
+  const {touched, error: formikError} = getFieldMeta(name);
```

Comment:
I'm not sure about using this `touched` state, because the email/phone number fields inside (indexed access) will also have an individual touched state.

So a touched with `{profile: true}` would be overwritten with `{touched: [true, false]}` by virtue of index access, I think.

Can we infer the field-as-a-whole touched state from the individual items touched state? And clicking the button to update preferences would then mark that item as touched too - I think that covers all possible situations.